lib/context.js

```diff
@@ -0,0 +1,81 @@
+const { execSync } = require('child_process');
+
+const { context } = require('@actions/github');
+const axios = require('axios');
+const ci = require('ci-info');
+
+/*
+ * This is an attempt to gather the same `context` regardless
+ * of where this code is being run (bitbucket or github actions)
+ *
+ * The general shape of the object we require is like this:
+ *
+ * ref: The fully-formed ref of the branch or tag that triggered the workflow run.
+ * sha: The commit SHA that triggered the workflow.
+ * actor: The name of the person or app that initiated the workflow.
+ * runId: A unique number for each workflow run within a repository.
+ * payload: The event webhook payload. Fetched from "GITHUB_EVENT_PATH"
+ *
+ * The payload needs to look like this:
+ * repository: { name: 'The name of the repo' }
+ * commits: [{ message: 'The commit message that triggered the workflow', author: { username: 'The username of the person who committed it' }}]
+ *
+ * https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+ *
+ * This is modelled off of the object that's returned from
+ * `require('@actions/github').context` and that's what we're
+ * using for github.
+ *
+ * Bitbucket doesn't have a module like this so we have to construct
+ * it ourselves based on what they do give us.
+ *
+ * https://support.atlassian.com/bitbucket-cloud/docs/variables-and-secrets/
+ */
+
+module.exports = async function getContext() {
+  switch (ci.name) {
+    case 'GitHub Actions':
+      return context;
+    case 'Bitbucket Pipelines': {
+      // Pipelines dont give us the username of the user who made the action
+      // happen, so we have to make an extra request to fetch this
+      // https://jira.atlassian.com/browse/BCLOUD-16711
+      const response = await axios.get(
+        `https://api.bitbucket.org/2.0/users/${process.env.BITBUCKET_STEP_TRIGGERER_UUID}`,
+        {
+          // We're okay with some level of error here
+          validateStatus(status) {
+            return status >= 200 && status < 500; // default
+          },
+        }
+      );
+
+      const actor = response.data.display_name || 'Unknown User';
+
+      return {
+        ref: execSync('git rev-parse --abbrev-ref HEAD', { encoding: 'utf-8' }).trim(),
+        sha: process.env.BITBUCKET_COMMIT,
+        actor,
+        runId: parseInt(process.env.BITBUCKET_BUILD_NUMBER, 10),
+        payload: {
+          repository: {
+            name: process.env.BITBUCKET_REPO_SLUG,
+          },
+          // commits: [
+          //   {
+          //     message: execSync('git log --format=%B -n 1 $BITBUCKET_COMMIT', {
```

Comment:
perhaps some of these are helpful references:
- https://github.com/infracost/infracost/blob/904f34aefdc9dbd393a4891060154a06b01ca7f2/internal/vcs/metadata.go#L821-L845
- https://github.com/ShiftLeftSecurity/sast-scan/blob/a4c6e6f21435e34c6bcc2e4dcc6e70682b8bce54/lib/context.py#L60
- https://github.com/danger/danger-js/blob/899a8200c0781b16d740c2d68a15e64530f6ab56/source/platforms/bitbucket_server/BitBucketServerAPI.ts#L112
- https://github.com/cypress-io/cypress/blob/bc6e9e49187aea3a8e021c39b5faee1059f16cf7/packages/server/lib/util/ci_provider.js#L177
- https://github.com/webdriverio/webdriverio/blob/f0c607340b4f31a4ae48509115335283f488e38f/packages/wdio-browserstack-service/src/util.ts#L320
- https://github.com/Shopify/cli/blob/1978fb22fbfa1c520edb70531bdd2acf07210ff5/packages/cli-kit/src/private/node/context/utilities.ts#L26
- https://github.com/rgerum/cameratransform/blob/c504311278d52929e3da0142a83c68ce04df81f5/coveralls.py#L42
- https://github.com/codecov/uploader/blob/859a1e73354939baa6bbd45eb8107084da9d62e9/src/ci_providers/provider_bitbucket.ts#L6
- https://github.com/DataDog/dd-trace-js/blob/6bb0d4ab38044b3559f7b3ac921acb4e73b3385f/packages/dd-trace/src/plugins/util/ci.js#L376

Reply:
Huh, i didnt even look. Would be quite a nice open source package to do this for all of the common CI providers. Like ci-info but more fleshed out https://www.npmjs.com/package/ci-info